odoo/tools/xml_utils.py

```diff
@@ -83,3 +85,47 @@ def create_xml_node(parent_node, node_name, node_value=None):
     :returns (etree._Element):
     """
     return create_xml_node_chain(parent_node, [node_name], node_value)[0]
+
+
+def load_xsd_from_url(url, module_name, env):
+    """Load an xsd file from given url and save it as ir.attachment
+
+    :param url: url of xsd file (str)
+    :param module_name: name of calling module (str)
+    :param env: odoo.api.Environment
+    :returns: saved attachment in ir.attachment or None if error
+    """
+    _logger = logging.getLogger(__name__)
+
+    fname = url.split('/')[-1]
+    xsd_fname = 'xsd_cached_%s' % fname.replace('.', '_')
+    attachment = env['ir.attachment'].search([('name', '=', xsd_fname)])
+    if attachment:
+        return attachment
```

Comment:
Seems optimistic and dangerous to assume schema files can not be updated.

Maybe fetching xsds should use more of a standard HTTP-type caching mechanism, follow caching directives it receives and not use the db as a cache implicitely?

Reply:
Maybe XSD validation could have something like an optional config key /  property, not set by default, and we could set that during tests (if there are any)?

This way institutional users who want to be really sure could enable validation on production (blocking the generation / output in case of invalidity).

Reply:
I understand that XSD validation should only be used in the development process, which is why only using it in tests seems like the way to go.
On the other hand, I also feel like it could be useful to let users know that the XML file they are downloading might not be valid (anymore), for instance if/when the XSD gets updated.

Reply:
> On the other hand, I also feel like it could be useful to let users know that the XML file they are downloading might not be valid (anymore), for instance if/when the XSD gets updated.

Per the way this is done in odoo/enterprise#23041 it looked to me like the user gets an error. So they don't "know that the file might not be valid", they are told it's not valid, apparently using undecorated errors straight from lxml, and they have no way to do or get anything.

Reply:
Yes, I know. But this is up for change (at least provide the user with clearer indication that the file is outdated).

I'm actually waiting to see what we plan on doing here (keep xsd validation on production - or not) before updating https://github.com/odoo/enterprise/pull/23041.

Reply:
OK, so I updated the method, cleaning up and making the caching optional. It is now intended to be mainly used for testing purposes.